cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManager.java

```diff
@@ -0,0 +1,226 @@
+                    alterQuotas(reconciliation, vertx, kafkaAdmin, clientQuotaAlteration)
+                        .onComplete(result -> {
+                            if (result.succeeded()) {
+                                LOGGER.infoCr(reconciliation, "Successfully altered default user quotas");
+                            } else {
+                                LOGGER.warnCr(reconciliation, "Failed to alter default user quotas", result.cause());
+                            }
+
+                            kafkaAdmin.close();
+                            promise.complete();
+                        });
+                } else {
+                    kafkaAdmin.close();
+                    promise.complete();
+                }
+
+                return promise.future();
+            });
+    }
+
+    /**
+     * Checks whether default quotas configuration needs to be altered with new configuration.
+     * It gets the current default quotas for users from Kafka and compares them with the desired configuration
+     *
+     * @param reconciliation          Reconciliation marker
+     * @param vertx                   Vert.x instance
+     * @param kafkaAdmin              Kafka Admin object
+     * @param ops                     List of {@link ClientQuotaAlteration.Op}
+     * @param isNotKafkaPlugin        boolean parameter determining if the Kafka built-in quotas plugin is used
+     *
+     * @return  result determining if the new quotas configuration should be applied or not
+     */
+    private static Future<Boolean> shouldAlterDefaultQuotasConfig(Reconciliation reconciliation, Vertx vertx, Admin kafkaAdmin, List<ClientQuotaAlteration.Op> ops, boolean isNotKafkaPlugin) {
+        return VertxUtil.kafkaFutureToVertxFuture(reconciliation,
+                vertx,
+                kafkaAdmin.describeClientQuotas(ClientQuotaFilter.containsOnly(List.of(ClientQuotaFilterComponent.ofDefaultEntity(ClientQuotaEntity.USER)))).entities())
+            .compose(clientQuotaEntityMapMap -> {
+                Map<String, Double> currentQuotas = clientQuotaEntityMapMap.get(DEFAULT_USER_ENTITY);
+
+                if (currentQuotas == null) {
+                    // 1. if the current quotas set in Kafka are null and quotas configuration in Kafka CR is not of type `QuotasPluginKafka`, skip alteration
+                    // 2. if the current quotas set in Kafka are null and quotas configuration in Kafka CR is type of `QuotasPluginKafka`, but there is no fields configured, skip alteration
+                    if (isNotKafkaPlugin || ops.stream().allMatch(op -> Objects.isNull(op.value()))) {
+                        LOGGER.debugCr(reconciliation, "There are no default user quotas set in Kafka and the Kafka built-in plugin is not configured, skipping the alteration");
+                        return Future.succeededFuture(false);
+                    } else if (ops.stream().anyMatch(op -> Objects.nonNull(op.value()))) {
```

Comment:
Do you really need the anyMatch here? Isn't it just else given you have the `allMatch` in the previous branch?

Reply:
Yeah I realized that a while back 😄 thanks